pkg/executor/content/fetcher.go

```diff
@@ -0,0 +1,112 @@
+package content
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+	"github.com/kubeshop/testkube/pkg/git"
+)
+
+func NewFetcher() Fetcher {
+	return Fetcher{}
+}
+
+type Fetcher struct {
+}
+
+func (f Fetcher) Fetch(content *testkube.ScriptContent) (path string, err error) {
+	switch testkube.ScriptContentType(content.Type_) {
+	case testkube.ScriptContentTypeFileURI:
+		return f.FetchURI(content.Uri)
+	case testkube.ScriptContentTypeString:
+		return f.FetchString(content.Data)
+	case testkube.ScriptContentTypeGitFile:
+		return f.FetchGitFile(content.Repository)
+	case testkube.ScriptContentTypeGitDir:
+		return f.FetchGitDir(content.Repository)
+	default:
+		return path, fmt.Errorf("unhandled content type: '%s'", content.Type_)
+	}
+}
+
+// FetchString stores string content as file
+func (f Fetcher) FetchString(str string) (path string, err error) {
+	return f.saveTempFile(strings.NewReader(str))
```

Comment:
Do we clean up this temp file somewhere?

Reply:
hmm, it's a very good question, probably it'll be wiped out when the job will end - as we don't store anything in any volumes (we've dropped volumes someday in past

Reply:
If we fetch data only in executor job, then yes, becuase it will be stored on container disk till container is alive